addons/account_peppol/models/res_partner.py

```diff
@@ -3,44 +3,66 @@
+
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    account_peppol_is_endpoint_valid = fields.Boolean(
-        string="PEPPOL endpoint validity",
-        help="The partner's EAS code and PEPPOL endpoint are valid",
-        compute="_compute_account_peppol_is_endpoint_valid", store=True,
-        copy=False,
-    )
-    account_peppol_validity_last_check = fields.Date(
-        string="Checked on",
-        help="Last Peppol endpoint verification",
-        compute="_compute_account_peppol_is_endpoint_valid", store=True,
-        copy=False,
-    )
-    account_peppol_verification_label = fields.Selection(
+    peppol_verification_state = fields.Selection(
         selection=[
             ('not_verified', 'Not verified yet'),
             ('not_valid', 'Not valid'),  # does not exist on Peppol at all
             ('not_valid_format', 'Cannot receive this format'),  # registered on Peppol but cannot receive the selected document type
             ('valid', 'Valid'),
         ],
-        string='Peppol endpoint validity',
-        compute='_compute_account_peppol_verification_label',
-        copy=False,
-    )  # field to compute the label to show for partner endpoint
+        string='Peppol endpoint verification',
+        default='not_verified',
+        company_dependent=True,
+        tracking=True,
+    )
     is_peppol_edi_format = fields.Boolean(compute='_compute_is_peppol_edi_format')
 
     # -------------------------------------------------------------------------
-    # BUSINESS ACTIONS
+    # OVERRIDES
+    # -------------------------------------------------------------------------
+
+    def _message_log(self, **kwargs):
+        peppol_verification_state_field = self.env['ir.model.fields']._get('res.partner', 'peppol_verification_state')
+        for tracked_value_id in kwargs.get('tracking_value_ids', []):
+            if tracked_value_id[2]['field_id'] == peppol_verification_state_field.id:
+                body = Markup("""
+                    <ul>
+                        <li>
+                            <span class='o-mail-Message-trackingOld me-1 px-1 text-muted fw-bold'>{old}</span>
+                            <i class='o-mail-Message-trackingSeparator fa fa-long-arrow-right mx-1 text-600'/>
+                            <span class='o-mail-Message-trackingNew me-1 fw-bold text-info'>{new}</span>
+                            <span class='o-mail-Message-trackingField ms-1 fst-italic text-muted'>({field})</span>
+                            <span class='o-mail-Message-trackingField ms-1 fst-italic text-muted'>({company})</span>
+                        </li>
+                    </ul>
+                """).format(
+                    old=tracked_value_id[2]['old_value_char'],
+                    new=tracked_value_id[2]['new_value_char'],
+                    field=peppol_verification_state_field.field_description,
+                    company=self.env.company.display_name,
```

Comment:
Why not logging yourself the message manually from:
`button_account_peppol_check_partner_endpoint`
`_update_peppol_state_per_company`
... to log the diff for all companies, not only the current one?